docs/multi-target-framework.md

```diff
@@ -0,0 +1,150 @@
+   Microsoft.OS.Sdk.net7.0_16.4. The corresponding branch will still produce a
+   Microsoft.NET.Sdk.iOS.Manifest-7.0.100 package, but that package shouldn't
+   be released/published.
+
+One idea might be to mark older release branches somehow, so they either don't
+produce these packages (only the ref/sdk/runtime packs), or alternatively they
+use a different package name in these branches.
+
+## Potential problems
+
+* MAX_PATH on Windows (the package names are longer by 10-11 characters).
+* Anything else?
+
+We'll select what's loaded at build time by doing this in WorkloadManifest.targets:
+
+```xml
+<ImportGroup Condition=" '$(TargetPlatformIdentifier)' == 'iOS' ">
+    <Import Project="Sdk.props"
+        Sdk="Microsoft.iOS.Sdk.net6"
+        Condition=" $([MSBuild]::VersionEquals($(TargetFrameworkVersion), '6.0')) " />
+    <Import Project="Sdk.props"
+        Sdk="Microsoft.iOS.Sdk.net7.0_16.1"
+        Condition=" $([MSBuild]::VersionEquals($(TargetFrameworkVersion), '7.0')) And '$(TargetPlatformVersion)' != '' And $([MSBuild]::VersionEquals($(TargetPlatformVersion), '16.1'))" />
+    <Import Project="Sdk.props"
+        Sdk="Microsoft.iOS.Sdk.net7.0_16.4"
+        Condition=" $([MSBuild]::VersionEquals($(TargetFrameworkVersion), '7.0')) And ('$(TargetPlatformVersion)' == '' Or $([MSBuild]::VersionEquals($(TargetPlatformVersion), '16.4')))" />
+    <Import Project="Sdk.props"
+        Sdk="Microsoft.iOS.Sdk.net8.0_17.0"
+        Condition=" $([MSBuild]::VersionEquals($(TargetFrameworkVersion), '8.0')) And '$(TargetPlatformVersion)' != '' And $([MSBuild]::VersionEquals($(TargetPlatformVersion), '17.0'))" />
+
+    <Import Project="Sdk-error.props"
+        Sdk="Microsoft.iOS.Sdk.net7.0_16.4"
+        Condition=" $([MSBuild]::VersionEquals($(TargetFrameworkVersion), '7.0')) And ('$(TargetPlatformVersion)' != '' And !$([MSBuild]::VersionEquals($(TargetPlatformVersion), '16.1') And !$([MSBuild]::VersionEquals($(TargetPlatformVersion), '16.4')))" />
+
+    <Import Project="Sdk-error.props"
+        Sdk="Microsoft.iOS.Sdk.net8.0_17.0"
+        Condition=" $([MSBuild]::VersionEquals($(TargetFrameworkVersion), '8.0')) And ('$(TargetPlatformVersion)' != '' And !$([MSBuild]::VersionEquals($(TargetPlatformVersion), '17.0')" />
+</ImportGroup>
+```
+
+Note 1: One complication here is that TargetPlatformVersion might not be set
+(if the TargetFramework doesn't contain the OS version), so there's one
+condition that accepts an empty TargetPlatformVersion. This corresponds with
+the default target platform version.
+
+Note 2: we load the preview sdk (Microsoft.iOS.Sdk.net8.0_17.0) even if
+`EnablePreviewFeatures!=true` - we show the error requesting
+`EnablePreviewFeatures` to be set from the preview sdk instead (this is to get
+an actionable error message). Without this, the user would get this rather
+unhelpful error message: `error NETSDK1139: The target platform identifier ios
+was not recognized.`
```

Comment:
Yeah, I wouldn't use this property to decide what to import. It's just a "signal" that this whole thing is preview and I have to acknowledge it.